Disable control flow e2e ADF

eromano opened the pull request:
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

<!--
 Before submitting your PR, please check that your code follows our contribution guidelines:
 https://github.com/Alfresco/alfresco-ng2-components/wiki/Code-contribution-acceptance-criteria
 -->

**What kind of change does this PR introduce?** (check one with "x")

> - [] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:


**What is the current behaviour?** (You can also link to an open issue here)
This PR disable the controll flow in ADF


**What is the new behaviour?**



**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No


If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

adinapitul commented:
1961 files changed?
shouldn't out-tsc folder be git ignored?

eromano commented:
> 1961 files changed?
> shouldn't out-tsc folder be git ignored?

Yep I will check

adinapitul commented on e2e/process-services-cloud/task-form-cloud-component.e2e.ts:
```diff
@@ -50,21 +50,31 @@ describe('Task form cloud component', () => {
+
+        const groupInfo = await groupIdentityService.getGroupInfoByGroupName('hr');
+        await identityService.addUserToGroup(testUser.idIdentityService, groupInfo.id);
+        await apiService.login(testUser.email, testUser.password);
```
duplicate call

eromano replied:
done

adinapitul commented on e2e/content-services/directives/create-library-directive.e2e.ts:
```diff
@@ -57,222 +57,221 @@ describe('Create library directive', function () {
-        createLibraryDialog.typeLibraryName(name);
+        await createLibraryDialog.typeLibraryName(name);
 
         for (let _i = 0; _i < 3; _i++) {
```
underscore can be removed

adinapitul commented on lib/testing/src/lib/process-services/pages/start-process-dialog.page.ts:
```diff
@@ -18,26 +18,27 @@
 import { BrowserVisibility } from '../../core/utils/browser-visibility';
 import { by, element } from 'protractor';
 import { BrowserActions } from '../../core/utils/browser-actions';
+import { ElementFinder } from 'protractor/built/element';
```
correct import would be from 'protractor' , though I don't think this exact one causes harm

adinapitul commented on lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts:
```diff
@@ -39,120 +39,119 @@ export class TaskListCloudComponentPage {
     }
 
-    clickCheckbox(taskName) {
+    clickCheckbox(taskName): Promise<void> {
```
needs async ?

eromano replied:
nope async needs to wrap te return in a promise this return already return a promise

adinapitul commented on lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts:
```diff
@@ -39,120 +39,119 @@ export class TaskListCloudComponentPage {
     }
 
-    checkRowIsNotChecked(taskName) {
+    checkRowIsNotChecked(taskName): Promise<void> {
```
async

eromano replied:
nope async needs to wrap te return in a promise this return already return a promise

adinapitul commented on lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts:
```diff
@@ -39,120 +39,119 @@ export class TaskListCloudComponentPage {
     }
 
-    checkRowIsChecked(taskName) {
+    checkRowIsChecked(taskName): Promise<void> {
```
async

eromano replied:
see above

adinapitul commented on lib/testing/src/lib/core/pages/form/widgets/displayTextWidget.ts:
```diff
@@ -16,23 +16,23 @@
+    inputLocator: Locator = by.css('input');
 
-    getFieldLabel(fieldId) {
+    getFieldLabel(fieldId): Promise<string> {
```
async for code consistency with all other pages

eromano replied:
see above

adinapitul commented on lib/testing/src/lib/core/pages/form/widgets/dateTimeWidget.ts:
```diff
@@ -16,81 +16,83 @@
     }
 
-    clearDateTimeInput(fieldId) {
+    async learDateTimeInput(fieldId): Promise<void> {
```
missing "c" from "clear" => duplicate correct method is at the end of the file

adinapitul commented on lib/testing/src/lib/process-services/pages/form-fields.page.ts:
```diff
@@ -16,144 +16,138 @@
 
-    isCompleteFormButtonDisabled() {
-        BrowserVisibility.waitUntilElementIsVisible(this.completeButton);
+    async isCompleteFormButtonDisabled(): Promise<string> {
```
according to the method name, I would expect a return type of Promise<boolean>

eromano replied:
not sure I got

adinapitul commented on lib/testing/src/lib/core/pages/form/formFields.ts:
```diff
@@ -15,149 +15,140 @@
 
-    isCompleteFormButtonDisabled() {
-        BrowserVisibility.waitUntilElementIsVisible(this.completeButton);
+    async isCompleteFormButtonDisabled(): Promise<string> {
```
according to the name, return type should be Promise<boolean>

eromano replied:
not sure I got is a promise

adinapitul replied:
sorry, should have escaped some chars, wanted to say that return type should be Promise of void, not string, according to the method name

adinapitul commented on e2e/pages/adf/viewerPage.ts:
```diff
@@ -15,666 +15,619 @@
 
-    checkFullScreenButtonIsNotDisplayed() {
-        BrowserVisibility.waitUntilElementIsNotVisible(this.fullScreenButton);
+    async checkFileContent(pageNumber, text): Promise<void> {
```
parameters types ?

adinapitul commented on e2e/pages/adf/viewerPage.ts:
```diff
@@ -15,666 +15,619 @@
-        return this;
+    async enterCustomName(text): Promise<void> {
+        const textField: ElementFinder = element(by.css('input[data-automation-id="adf-text-custom-name"]'));
+        await BrowserVisibility.waitUntilElementIsVisible(textField);
```
not needed. wait for visibility already done in clearSendKeys

adinapitul commented on e2e/pages/adf/process-services/processServiceTabBarPage.ts:
```diff
@@ -16,33 +16,34 @@
-        BrowserActions.click(this.tasksButton);
-        browser.sleep(700);
+    async clickTasksButton(): Promise<void> {
+        await BrowserActions.closeMenuAndDialogs();
```
closing other possible left over dialogs should be the responsibility of the test
this method should do what the name says, just click on the Tasks button

adinapitul commented on e2e/pages/adf/contentServicesPage.ts:
```diff
@@ -35,166 +35,158 @@ export class ContentServicesPage {
-        const actionButton = element(by.css(`button[data-automation-id="context-${actionName}"`));
-        BrowserVisibility.waitUntilElementIsVisible(actionButton);
-        return actionButton.isEnabled();
+    async checkContextActionIsEnabled(actionName): Promise<boolean> {
```
name is misleading, suggest: isContextActionEnabled

adinapitul commented on e2e/pages/adf/contentServicesPage.ts:
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
-        this.contentList.doubleClickRow(nodeName);
-        return this;
+    async doubleClickRow(nodeName): Promise<void> {
+        await BrowserActions.closeMenuAndDialogs();
```
this should be the responsibility of the test
this method should have a single responsibility: double clicking on an element.

adinapitul commented on e2e/pages/adf/contentServicesPage.ts:
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
 
-    checkEditFolderButtonIsEnabled() {
-        return this.editFolderButton.isEnabled();
+    async checkEditFolderButtonIsEnabled(): Promise<boolean> {
```
name doesn't match return type, suggest: isEditFolderButtonEnabled

adinapitul commented on e2e/pages/adf/contentServicesPage.ts:
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
-        return this;
+    async getErrorMessage(): Promise<string> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.errorSnackBar);
+        return await this.errorSnackBar.getText();
```
BrowserActions could be used here

adinapitul commented on e2e/pages/adf/contentServicesPage.ts:
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
-        const optionToClick = element(by.css(`mat-option[data-automation-id="grid-view-sorting-${sortingChosen}"]`));
-        BrowserActions.click(optionToClick);
+    async selectGridSortingFromDropdown(sortingChosen): Promise<void> {
+        // await BrowserActions.closeMenuAndDialogs();
```
remove comment if not needed

eromano replied:
from the blame has been committed by you , I usually not comment code ;)

adinapitul replied:
true, none the less, should be removed :D

adinapitul commented on e2e/pages/adf/content-services/breadcrumb/breadCrumbDropdownPage.ts:
```diff
@@ -15,32 +15,32 @@
-    checkBreadCrumbDropdownIsDisplayed() {
-        BrowserVisibility.waitUntilElementIsVisible(this.breadCrumbDropdown);
+    async checkBreadCrumbDropdownIsDisplayed(): Promise<void> {
+        await BrowserVisibility. waitUntilElementIsVisible(this.breadCrumbDropdown);
```
extra space

adinapitul commented on e2e/pages/adf/configEditorPage.ts:
```diff
@@ -15,35 +15,33 @@
-        return this;
+    async enterConfiguration(text): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.textField);
+        await this.textField.sendKeys(text);
```
BrowserActions could be used

adinapitul commented on e2e/pages/adf/commentsPage.ts:
```diff
@@ -15,59 +15,59 @@
-    checkUserIconIsDisplayed(position) {
-        BrowserVisibility.waitUntilElementIsVisible(this.commentUserIcon.first());
+    async checkUserIconIsDisplayed(position): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.commentUserIcon.first());
```
the return on this method doesn't make sense.

adinapitul commented on e2e/pages/adf/commentsPage.ts:
```diff
@@ -15,59 +15,59 @@
-        BrowserActions.click(this.addCommentButton);
+    async addComment(comment): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.commentInput);
+        await this.commentInput.sendKeys(comment);
```
BrowserActions could be used

adinapitul commented on e2e/pages/adf/cardViewComponentPage.ts:
```diff
@@ -15,182 +15,166 @@
-        return this;
+    async setName(name): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.nameInputField);
+        await this.nameInputField.sendKeys(name);
```
BrowserActions could be used

adinapitul commented on e2e/content-services/document-list/document-list-gallery.e2e.ts:
```diff
@@ -37,7 +37,7 @@ describe('Document List Component', () => {
     const navBar = new NavigationBarPage();
```
do we really need 2 NavigationBarPages, navBar and navigationBarPage ?

adinapitul commented on e2e/content-services/document-list/document-list-permissions.e2e.ts:
```diff
@@ -33,16 +33,16 @@ describe('Document List Component', () => {
     let acsUser = null;
 
-    beforeAll(() => {
+    beforeAll(async () => {
```
async not needed

adinapitul commented on e2e/content-services/document-list/document-list-component.e2e.ts:
```diff
@@ -346,54 +345,54 @@ describe('Document List Component', () => {
-            Promise.all(folderCreated.map((folder) =>
-                uploadActions.deleteFileOrFolder(folder.entry.id)
+        afterAll(async () => {
+            await Promise.all(folderCreated.map(async (folder) =>
```
for..of might be safer to use ?

eromano commented on e2e/pages/adf/process-services/tasksListPage.ts:
```diff
@@ -17,48 +17,47 @@
     }
 
-    getRowsDisplayedWithSameName(taskName) {
+    getRowsDisplayedWithSameName(taskName): Promise<string> {
```
see above

adinapitul commented on e2e/core/login/login-sso/login-sso.e2e.ts:
```diff
@@ -18,6 +18,7 @@
 import { LoginSSOPage, SettingsPage, LoginPage } from '@alfresco/adf-testing';
 import { browser } from 'protractor';
 import { NavigationBarPage } from '../../../pages/adf/navigationBarPage';
+import { BrowserVisibility } from '../../../../lib/testing/src/lib/core/utils/browser-visibility';
```
should be '@alfresco/adf-testing'

eromano commented:
@adinapitul I have fixed all the comments except the one about async. see my comment. Some of your comment that I have implemented I guess where anyway not in the scope of this PR but they were more a name refactoring. I am ok with it but for now, I would like to postpone further improvements in a separate PR

adinapitul commented:
regarding those asyncs... Take the following for example:
`    checkContentIsNotDisplayedByName(taskName): Promise<void> {
        return this.dataTable.checkContentIsNotDisplayed(column.name, taskName);
    }`
at this moment async indeed is not needed since there's no await inside.
which got me thinking as to WHY there's no await inside.
I think this method should look like this:
`async checkContentIsNotDisplayedByName(taskName): Promise<void> {
        await this.dataTable.checkContentIsNotDisplayed(column.name, taskName);
    }`
cause it doesn't really make sense to have return there.

adinapitul commented:
to follow on my previous comment, what's the difference between
this: https://github.com/Alfresco/alfresco-ng2-components/blob/f4a0a4bc9320598418f697e54e3ae4e50d428ff9/lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts#L78
and this: https://github.com/Alfresco/alfresco-ng2-components/blob/f4a0a4bc9320598418f697e54e3ae4e50d428ff9/lib/testing/src/lib/content-services/pages/document-list.page.ts#L57 
?
we need to be consistent
